crates/core/src/transport/connection_handler.rs

```diff
@@ -250,7 +244,11 @@ impl UdpPacketsListener {
                 }
                 ConnectionState::AckConnection => {
                     self.socket
-                        .send_to(hello_packet.send_data(), remote_addr)
+                        .send_to(
+                            SymmetricMessage::ack_ok(outbound_sym_key.as_mut().unwrap())?
```

Comment:
Why is outbound_sym_key borrowed as mutable here?

Reply:
This is a reminder of when you were not using encrypt in place since that trait took a &mut instead